Index: kdeplasma-addons-4.1.2/dataengines/comic/CMakeLists.txt
===================================================================
--- kdeplasma-addons-4.1.2.orig/dataengines/comic/CMakeLists.txt	2008-09-26 22:09:01.000000000 +0200
+++ kdeplasma-addons-4.1.2/dataengines/comic/CMakeLists.txt	2008-09-26 22:11:10.000000000 +0200
@@ -71,7 +71,7 @@
     )
 
     kde4_add_plugin( plasma_comic_osnewsprovider ${comic_osnews_provider_SRCS} )
-    target_link_libraries( plasma_comic_osnewsprovider plasmacomicprovidercore ${KDE4_SYNDICATION_LIBS} )
+    target_link_libraries( plasma_comic_osnewsprovider plasmacomicprovidercore ${KDE4_SYNDICATION_LIBS} ${QT_QTNETWORK_LIBRARY} )
     install( TARGETS plasma_comic_osnewsprovider DESTINATION ${PLUGIN_INSTALL_DIR} )
     install( FILES osnewsprovider.desktop DESTINATION ${SERVICES_INSTALL_DIR} )
     install( FILES pics/plasma_comic_osnews.png DESTINATION ${DATA_INSTALL_DIR}/plasma-comic )
@@ -83,7 +83,7 @@
     )
 
     kde4_add_plugin( plasma_comic_snoopyprovider ${comic_snoopy_provider_SRCS} )
-    target_link_libraries( plasma_comic_snoopyprovider plasmacomicprovidercore ${QT_QTGUI_LIBRARY} ${KDE4_KIO_LIBS} )
+    target_link_libraries( plasma_comic_snoopyprovider plasmacomicprovidercore ${QT_QTGUI_LIBRARY} ${KDE4_KIO_LIBS} ${QT_QTNETWORK_LIBRARY} )
     install( TARGETS plasma_comic_snoopyprovider DESTINATION ${PLUGIN_INSTALL_DIR} )
     install( FILES snoopyprovider.desktop DESTINATION ${SERVICES_INSTALL_DIR} )
     install( FILES pics/plasma_comic_snoopy.png DESTINATION ${DATA_INSTALL_DIR}/plasma-comic )
@@ -95,7 +95,7 @@
     )
 
     kde4_add_plugin( plasma_comic_xkcdprovider ${comic_xkcd_provider_SRCS} )
-    target_link_libraries( plasma_comic_xkcdprovider plasmacomicprovidercore ${KDE4_KIO_LIBS} ${QT_QTGUI_LIBRARY} )
+    target_link_libraries( plasma_comic_xkcdprovider plasmacomicprovidercore ${KDE4_KIO_LIBS} ${QT_QTGUI_LIBRARY} ${QT_QTNETWORK_LIBRARY} )
     install( TARGETS plasma_comic_xkcdprovider DESTINATION ${PLUGIN_INSTALL_DIR} )
     install( FILES xkcdprovider.desktop DESTINATION ${SERVICES_INSTALL_DIR} )
     install( FILES pics/plasma_comic_xkcd.png DESTINATION ${DATA_INSTALL_DIR}/plasma-comic )
@@ -107,7 +107,7 @@
     )
 
     kde4_add_plugin( plasma_comic_phdprovider ${comic_phd_provider_SRCS} )
-    target_link_libraries( plasma_comic_phdprovider plasmacomicprovidercore ${QT_QTGUI_LIBRARY} ${KDE4_KIO_LIBS} )
+    target_link_libraries( plasma_comic_phdprovider plasmacomicprovidercore ${QT_QTGUI_LIBRARY} ${KDE4_KIO_LIBS} ${QT_QTNETWORK_LIBRARY} )
     install( TARGETS plasma_comic_phdprovider DESTINATION ${PLUGIN_INSTALL_DIR} )
     install( FILES phdprovider.desktop DESTINATION ${SERVICES_INSTALL_DIR} )
     install( FILES pics/plasma_comic_phd.png DESTINATION ${DATA_INSTALL_DIR}/plasma-comic )
