--- a/dataengines/comic/CMakeLists.txt
+++ b/dataengines/comic/CMakeLists.txt
@@ -68,7 +68,7 @@
         osnewsprovider.cpp
    )
    kde4_add_plugin( plasma_comic_osnewsprovider ${comic_osnews_provider_SRCS} )
-   target_link_libraries( plasma_comic_osnewsprovider plasmacomicprovidercore ${KDE4_SYNDICATION_LIBS} )
+   target_link_libraries( plasma_comic_osnewsprovider plasmacomicprovidercore ${KDE4_SYNDICATION_LIBS} ${QT_QTNETWORK_LIBRARY} )
    install( TARGETS plasma_comic_osnewsprovider DESTINATION ${PLUGIN_INSTALL_DIR} )
    install( FILES osnewsprovider.desktop DESTINATION ${SERVICES_INSTALL_DIR} )
    install( FILES pics/plasma_comic_osnews.png DESTINATION ${DATA_INSTALL_DIR}/plasma-comic )
@@ -81,7 +81,7 @@
 )
 
 kde4_add_plugin( plasma_comic_snoopyprovider ${comic_snoopy_provider_SRCS} )
-target_link_libraries( plasma_comic_snoopyprovider plasmacomicprovidercore ${KDE4_SYNDICATION_LIBS} )
+target_link_libraries( plasma_comic_snoopyprovider plasmacomicprovidercore ${KDE4_SYNDICATION_LIBS} ${QT_QTNETWORK_LIBRARY} )
 install( TARGETS plasma_comic_snoopyprovider DESTINATION ${PLUGIN_INSTALL_DIR} )
 install( FILES snoopyprovider.desktop DESTINATION ${SERVICES_INSTALL_DIR} )
 install( FILES pics/plasma_comic_snoopy.png DESTINATION ${DATA_INSTALL_DIR}/plasma-comic )
@@ -93,7 +93,7 @@
 )
 
 kde4_add_plugin( plasma_comic_xkcdprovider ${comic_xkcd_provider_SRCS} )
-target_link_libraries( plasma_comic_xkcdprovider plasmacomicprovidercore ${KDE4_SYNDICATION_LIBS} )
+target_link_libraries( plasma_comic_xkcdprovider plasmacomicprovidercore ${KDE4_SYNDICATION_LIBS}  ${QT_QTNETWORK_LIBRARY})
 install( TARGETS plasma_comic_xkcdprovider DESTINATION ${PLUGIN_INSTALL_DIR} )
 install( FILES xkcdprovider.desktop DESTINATION ${SERVICES_INSTALL_DIR} )
 install( FILES pics/plasma_comic_xkcd.png DESTINATION ${DATA_INSTALL_DIR}/plasma-comic )
@@ -105,7 +105,7 @@
 )
 
 kde4_add_plugin( plasma_comic_phdprovider ${comic_phd_provider_SRCS} )
-target_link_libraries( plasma_comic_phdprovider plasmacomicprovidercore ${KDE4_SYNDICATION_LIBS} )
+target_link_libraries( plasma_comic_phdprovider plasmacomicprovidercore ${KDE4_SYNDICATION_LIBS} ${QT_QTNETWORK_LIBRARY} )
 install( TARGETS plasma_comic_phdprovider DESTINATION ${PLUGIN_INSTALL_DIR} )
 install( FILES phdprovider.desktop DESTINATION ${SERVICES_INSTALL_DIR} )
 install( FILES pics/plasma_comic_phd.png DESTINATION ${DATA_INSTALL_DIR}/plasma-comic )
