Index: kdeplasma-addons-4.11.2/applets/news/plasma-applet-news.desktop
===================================================================
--- kdeplasma-addons-4.11.2.orig/applets/news/plasma-applet-news.desktop	2013-09-27 17:22:45.752875933 +0100
+++ kdeplasma-addons-4.11.2/applets/news/plasma-applet-news.desktop	2013-09-27 17:22:45.744875933 +0100
@@ -118,7 +118,7 @@
 X-KDE-Library=plasma_applet_news
 X-KDE-PluginInfo-Author=Petri Damstén
 X-KDE-PluginInfo-Email=damu@iki.fi
-X-KDE-PluginInfo-Name=news
+X-KDE-PluginInfo-Name=org.kde.news-qml
 X-KDE-PluginInfo-Version=1.0
 X-KDE-PluginInfo-Website=http://plasma.kde.org/
 X-KDE-PluginInfo-Category=Online Services
