Index: kdeplasma-addons-4.2.85/applets/bball/bball.h
===================================================================
--- kdeplasma-addons-4.2.85.orig/applets/bball/bball.h	2009-05-16 16:17:45.000000000 -0400
+++ kdeplasma-addons-4.2.85/applets/bball/bball.h	2009-05-16 16:18:02.000000000 -0400
@@ -32,7 +32,7 @@
 #include <Phonon/MediaObject>
 #include <Phonon/Path>
 #include <Phonon/AudioOutput>
-#include <Phonon/Global>
+#include <Phonon/phononnamespace.h>
 
 #include "ui_bballConfig.h"
 
