Index: kdeplasma-addons-4.5.2/applets/CMakeLists.txt
===================================================================
--- kdeplasma-addons-4.5.2.orig/applets/CMakeLists.txt	2010-10-22 22:06:11.600971000 +0200
+++ kdeplasma-addons-4.5.2/applets/CMakeLists.txt	2010-10-22 22:09:25.512971000 +0200
@@ -83,11 +83,11 @@
 if (Q_WS_X11)
    if (X11_Xrender_FOUND)
       set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${CMAKE_CURRENT_SOURCE_DIR}/kimpanel/cmake)
-      macro_optional_find_package(SCIM)
-      macro_log_feature(SCIM_FOUND "SCIM headers" "Multibyte input widget" "http://www.scim-im.org" FALSE "1.4" "Needed for the kimpanel multibyte input Plasmoid")
-      if (SCIM_FOUND)
+#      macro_optional_find_package(SCIM)
+#      macro_log_feature(SCIM_FOUND "SCIM headers" "Multibyte input widget" "http://www.scim-im.org" FALSE "1.4" "Needed for the kimpanel multibyte input Plasmoid")
+#      if (SCIM_FOUND)
          add_subdirectory(kimpanel)
-      endif ( SCIM_FOUND)
+#      endif ( SCIM_FOUND)
    endif (X11_Xrender_FOUND)
 endif (Q_WS_X11)
 
Index: kdeplasma-addons-4.5.2/applets/kimpanel/backend/CMakeLists.txt
===================================================================
--- kdeplasma-addons-4.5.2.orig/applets/kimpanel/backend/CMakeLists.txt	2010-10-22 22:10:05.068971002 +0200
+++ kdeplasma-addons-4.5.2/applets/kimpanel/backend/CMakeLists.txt	2010-10-22 22:10:16.380971002 +0200
@@ -1 +1 @@
-macro_optional_add_subdirectory(scim)
+#macro_optional_add_subdirectory(scim)
