diff -rupN 4.3-kdeplasma-addons/applets/rememberthemilk/authenticate.ui trunk-kdeplasma-addons/applets/rememberthemilk/authenticate.ui
--- 4.3-kdeplasma-addons/applets/rememberthemilk/authenticate.ui	2009-10-10 11:00:29.000000000 -0400
+++ trunk-kdeplasma-addons/applets/rememberthemilk/authenticate.ui	2009-10-10 11:06:42.000000000 -0400
@@ -47,44 +47,25 @@
    <item>
     <widget class="QGroupBox" name="groupBox">
      <property name="title">
-      <string>Authenticate as:</string>
+      <string>Re-Authenticate</string>
      </property>
      <property name="flat">
       <bool>true</bool>
      </property>
      <layout class="QFormLayout" name="formLayout">
-      <item row="0" column="0">
-       <widget class="QLabel" name="label">
-        <property name="text">
-         <string>Username:</string>
-        </property>
-       </widget>
-      </item>
+      <property name="fieldGrowthPolicy">
+       <enum>QFormLayout::ExpandingFieldsGrow</enum>
+      </property>
+      <property name="labelAlignment">
+       <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
+      </property>
+      <property name="formAlignment">
+       <set>Qt::AlignHCenter|Qt::AlignTop</set>
+      </property>
       <item row="0" column="1">
-       <widget class="KLineEdit" name="username">
-        <property name="whatsThis">
-         <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
-&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
-p, li { white-space: pre-wrap; }
-&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Segoe UI'; font-size:9pt; font-weight:400; font-style:normal;&quot;&gt;
-&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;Enter your Remember The Milk username here. If you do not have an account, visit &lt;a href=&quot;http://www.rememberthemilk.com/signup/&quot;&gt;&lt;span style=&quot; text-decoration: underline; color:#0000ff;&quot;&gt;http://www.rememberthemilk.com/signup/&lt;/span&gt;&lt;/a&gt; to create one.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
-        </property>
-       </widget>
-      </item>
-      <item row="1" column="0">
-       <widget class="QLabel" name="label_2">
+       <widget class="QPushButton" name="authenticate">
         <property name="text">
-         <string>Password:</string>
-        </property>
-       </widget>
-      </item>
-      <item row="1" column="1">
-       <widget class="KLineEdit" name="password">
-        <property name="whatsThis">
-         <string>Enter your Remember The Milk password here.</string>
-        </property>
-        <property name="passwordMode">
-         <bool>true</bool>
+         <string>Authenticate with Remember The Milk Service</string>
         </property>
        </widget>
       </item>
@@ -136,17 +117,12 @@ p, li { white-space: pre-wrap; }
  <customwidgets>
   <customwidget>
    <class>KLed</class>
-   <extends></extends>
+   <extends>QWidget</extends>
    <header>kled.h</header>
   </customwidget>
   <customwidget>
-   <class>KLineEdit</class>
-   <extends></extends>
-   <header>klineedit.h</header>
-  </customwidget>
-  <customwidget>
    <class>KUrlLabel</class>
-   <extends></extends>
+   <extends>QLabel</extends>
    <header>kurllabel.h</header>
   </customwidget>
  </customwidgets>
diff -rupN 4.3-kdeplasma-addons/applets/rememberthemilk/rememberthemilk-plasmoid.cpp trunk-kdeplasma-addons/applets/rememberthemilk/rememberthemilk-plasmoid.cpp
--- 4.3-kdeplasma-addons/applets/rememberthemilk/rememberthemilk-plasmoid.cpp	2009-10-10 11:00:29.000000000 -0400
+++ trunk-kdeplasma-addons/applets/rememberthemilk/rememberthemilk-plasmoid.cpp	2009-10-10 11:06:42.000000000 -0400
@@ -20,23 +20,12 @@
 #include "rememberthemilk-plasmoid.h"
 
 // Qt Includes
-#include <QApplication>
-#include <QPainter>
-#include <QFontMetrics>
-#include <QSizeF>
-#include <QStringListModel>
-#include <QTreeView>
-#include <QHeaderView>
-#include <QTimer>
-#include <QVBoxLayout>
-#include <QWebView>
 #include <QPushButton>
 #include <QCheckBox>
+#include <QTreeView>
 
 // KDE Includes
-#include <KTitleWidget>
 #include <KConfigDialog>
-#include <KLineEdit>
 #include <KDebug> 
 #include <KToolInvocation>
 
@@ -55,6 +44,7 @@
 #include "ui_general.h"
 #include "taskmodel.h"
 #include "tasksortfilter.h"
+#include <KLineEdit>
 
 RememberTheMilkPlasmoid::RememberTheMilkPlasmoid(QObject* parent, const QVariantList& args)
     : Plasma::PopupApplet(parent, args),
@@ -125,16 +115,16 @@ void RememberTheMilkPlasmoid::init() {
   Plasma::Applet::init();
 }
 
+
+void RememberTheMilkPlasmoid::startAuth()
+{
+  KConfigGroup cg = m_authService->operationDescription("Login");
+  busyUntil(m_authService->startOperationCall(cg));
+  busyUntil(0); // Sets busy until we manually call jobFinished(0). Busy until first tasks refresh
+  m_authenticated = false;  
+}
+
 void RememberTheMilkPlasmoid::configAccepted() {
-  if (!m_authWidgetUi->username->text().isEmpty()) {
-      // Remove/replace tabs?
-      KConfigGroup cg = m_authService->operationDescription("Login");
-      cg.writeEntry("username", m_authWidgetUi->username->text());
-      cg.writeEntry("password", m_authWidgetUi->password->text());
-      busyUntil(m_authService->startOperationCall(cg));
-      busyUntil(0); // Sets busy until we manually call jobFinished(0). Busy until first tasks refresh
-      m_authenticated = false;
-  }
   switch(m_generalOptionsUi->sortType->currentIndex()) {
     case 0:
       setSortBy(SortDue);
@@ -155,18 +145,7 @@ void RememberTheMilkPlasmoid::createConf
   connect(parent, SIGNAL(finished()), this, SLOT(configFinished()));
   connect(parent, SIGNAL(applyClicked()), this, SLOT(configAccepted()));
   connect(parent, SIGNAL(okClicked()), this, SLOT(configAccepted()));
-  
-  if (m_authenticated) {
-    m_authWidgetUi->authStatus->setText(i18n("Authenticated"));
-    m_authWidgetUi->kled->setState(KLed::On);
-    m_authWidgetUi->kled->setColor(Qt::green);
-  } else {
-    m_authWidgetUi->authStatus->setText(i18n("Not Authenticated"));
-    m_authWidgetUi->kled->setState(KLed::Off); 
-    m_authWidgetUi->kled->setColor(Qt::red);
-  }
-  m_authWidgetUi->username->clear();
-  m_authWidgetUi->password->clear();
+  connect(m_authWidgetUi->authenticate, SIGNAL(clicked(bool)), this, SLOT(startAuth()));
   
   m_generalOptionsUi->sortType->setCurrentIndex(m_sortBy);
   
@@ -201,6 +180,17 @@ void RememberTheMilkPlasmoid::dataUpdate
   if (name == "Auth") {
     m_authenticated = data.value("ValidToken").toBool();
     kDebug() << "Auth: " << m_authenticated;
+    
+    if (m_authenticated) {
+      m_authWidgetUi->authStatus->setText(i18n("Authenticated"));
+      m_authWidgetUi->kled->setState(KLed::On);
+      m_authWidgetUi->kled->setColor(Qt::green);
+    } else {
+      m_authWidgetUi->authStatus->setText(i18n("Not Authenticated"));
+      m_authWidgetUi->kled->setState(KLed::Off); 
+      m_authWidgetUi->kled->setColor(Qt::red);
+    }
+    
     if (m_authenticated) {
       setConfigurationRequired(false);
       m_token = data.value("Token").toString();
diff -rupN 4.3-kdeplasma-addons/applets/rememberthemilk/rememberthemilk-plasmoid.h trunk-kdeplasma-addons/applets/rememberthemilk/rememberthemilk-plasmoid.h
--- 4.3-kdeplasma-addons/applets/rememberthemilk/rememberthemilk-plasmoid.h	2009-10-10 11:00:29.000000000 -0400
+++ trunk-kdeplasma-addons/applets/rememberthemilk/rememberthemilk-plasmoid.h	2009-10-10 11:06:42.000000000 -0400
@@ -23,9 +23,6 @@
 #include <Plasma/PopupApplet>
 
 #include <QGraphicsLinearLayout>
-#include <QStringListModel>
-#include <QStandardItemModel>
-#include <QListView>
 #include <QList>
 
 #include "taskitem.h"
@@ -74,6 +71,7 @@ class RememberTheMilkPlasmoid : public P
     void createConfigurationInterface(KConfigDialog *parent);
     void configAccepted();
     void configFinished();
+    void startAuth();
     
     void setSortBy(SortBy sortBy);
     
diff -rupN 4.3-kdeplasma-addons/applets/rememberthemilk/taskitem.cpp trunk-kdeplasma-addons/applets/rememberthemilk/taskitem.cpp
--- 4.3-kdeplasma-addons/applets/rememberthemilk/taskitem.cpp	2009-10-10 11:00:29.000000000 -0400
+++ trunk-kdeplasma-addons/applets/rememberthemilk/taskitem.cpp	2009-10-10 11:06:42.000000000 -0400
@@ -19,8 +19,6 @@
 
 #include "taskitem.h"
 
-#include <QPainter>
-#include <QStyleOption>
 
 #include <KDebug>
 
diff -rupN 4.3-kdeplasma-addons/applets/rememberthemilk/taskitemdelegate.cpp trunk-kdeplasma-addons/applets/rememberthemilk/taskitemdelegate.cpp
--- 4.3-kdeplasma-addons/applets/rememberthemilk/taskitemdelegate.cpp	2009-10-10 11:00:29.000000000 -0400
+++ trunk-kdeplasma-addons/applets/rememberthemilk/taskitemdelegate.cpp	2009-10-10 11:06:42.000000000 -0400
@@ -29,8 +29,6 @@
 #include <QColor>
 #include <QLinearGradient>
 #include <QCheckBox>
-#include <QAbstractItemView>
-#include <QDateTime>
 
 #include <KDebug>
 
@@ -112,9 +110,18 @@ void TaskItemDelegate::paintPriorityHead
   QLinearGradient gradient(rect.topLeft(), rect.bottomRight());
   gradient.setColorAt(0, itemPriorityColor(index));
   gradient.setColorAt(1, Qt::transparent);
-  painter->setBrush(gradient);
-  painter->setPen(Qt::NoPen);
-  painter->drawRect(rect);
+  
+  int h = QApplication::fontMetrics().height();
+  QPainterPath path;
+  path.moveTo(rect.bottomLeft());
+  path.quadTo(rect.bottomLeft()+QPointF(0, -h), rect.bottomLeft()+QPointF(2, -h));
+  path.lineTo(rect.bottomRight()-QPointF(2, h));
+  path.quadTo(rect.bottomRight()+QPointF(0, -h), rect.bottomRight());
+  painter->setBrush(Qt::NoBrush);
+  QPen thickPen(itemPriorityColor(index));
+  thickPen.setWidth(2);
+  painter->setPen(thickPen);
+  painter->drawPath(path);
   
   // Draw priority text
   QString priority = index.data(Qt::DisplayRole).toString();
diff -rupN 4.3-kdeplasma-addons/applets/rememberthemilk/tasksortfilter.cpp trunk-kdeplasma-addons/applets/rememberthemilk/tasksortfilter.cpp
--- 4.3-kdeplasma-addons/applets/rememberthemilk/tasksortfilter.cpp	2009-10-10 11:00:29.000000000 -0400
+++ trunk-kdeplasma-addons/applets/rememberthemilk/tasksortfilter.cpp	2009-10-10 11:06:42.000000000 -0400
@@ -20,7 +20,6 @@
 #include "tasksortfilter.h"
 #include "taskitem.h"
 #include "taskmodel.h"
-#include <QDateTime>
 
 
 TaskSortFilter::TaskSortFilter(TaskModel *model, QObject* parent)
diff -rupN 4.3-kdeplasma-addons/dataengines/rememberthemilk/authservice.cpp trunk-kdeplasma-addons/dataengines/rememberthemilk/authservice.cpp
--- 4.3-kdeplasma-addons/dataengines/rememberthemilk/authservice.cpp	2009-10-10 11:00:19.000000000 -0400
+++ trunk-kdeplasma-addons/dataengines/rememberthemilk/authservice.cpp	2009-10-10 11:06:30.000000000 -0400
@@ -48,7 +48,7 @@ void AuthJob::start() {
   connect(m_session, SIGNAL(tokenCheck(bool)), SLOT(result(bool)));
   //FIXME: error handling?
   if (operationName() == "Login") {
-    m_session->login(parameters().value("username").toString(), parameters().value("password").toString());
+    m_session->showLoginWindow();
   } 
   else if (operationName() == "AuthWithToken") {
     m_session->setToken(parameters().value("token").toString());
diff -rupN 4.3-kdeplasma-addons/dataengines/rememberthemilk/authservice.h trunk-kdeplasma-addons/dataengines/rememberthemilk/authservice.h
--- 4.3-kdeplasma-addons/dataengines/rememberthemilk/authservice.h	2009-10-10 11:00:19.000000000 -0400
+++ trunk-kdeplasma-addons/dataengines/rememberthemilk/authservice.h	2009-10-10 11:06:30.000000000 -0400
@@ -24,7 +24,6 @@
 #include <Plasma/ServiceJob>
 
 namespace RTM {
-  class Request;
   class Session;
 }
 
diff -rupN 4.3-kdeplasma-addons/dataengines/rememberthemilk/listssource.h trunk-kdeplasma-addons/dataengines/rememberthemilk/listssource.h
--- 4.3-kdeplasma-addons/dataengines/rememberthemilk/listssource.h	2009-10-10 11:00:19.000000000 -0400
+++ trunk-kdeplasma-addons/dataengines/rememberthemilk/listssource.h	2009-10-10 11:06:30.000000000 -0400
@@ -22,8 +22,6 @@
 
 #include <rtm/rtm.h>
 
-#include <QList>
-#include <QPair>
 #include <QTimer>
 
 #include <Plasma/DataContainer>
diff -rupN 4.3-kdeplasma-addons/dataengines/rememberthemilk/rtmauth.operations trunk-kdeplasma-addons/dataengines/rememberthemilk/rtmauth.operations
--- 4.3-kdeplasma-addons/dataengines/rememberthemilk/rtmauth.operations	2009-10-10 11:00:19.000000000 -0400
+++ trunk-kdeplasma-addons/dataengines/rememberthemilk/rtmauth.operations	2009-10-10 11:06:30.000000000 -0400
@@ -3,12 +3,6 @@
     "http://www.kde.org/standards/kcfg/1.0/kcfg.xsd">
 <kcfg>
     <group name="Login">
-        <entry name="username" type="String">
-          <label>The username of the user to authenticate as</label>
-        </entry>
-        <entry name="password" type="Password">
-          <label>The password associated with the account</label>
-        </entry>
     </group>
     <group name="AuthWithToken">
         <entry name="token" type="String">
diff -rupN 4.3-kdeplasma-addons/dataengines/rememberthemilk/rtmengine.cpp trunk-kdeplasma-addons/dataengines/rememberthemilk/rtmengine.cpp
--- 4.3-kdeplasma-addons/dataengines/rememberthemilk/rtmengine.cpp	2009-10-10 11:00:19.000000000 -0400
+++ trunk-kdeplasma-addons/dataengines/rememberthemilk/rtmengine.cpp	2009-10-10 11:06:30.000000000 -0400
@@ -26,19 +26,16 @@
 #include "authservice.h"
 #include "tasksservice.h"
 
-#include <QDate>
-#include <QTime>
  
-#include <KSystemTimeZones>
-#include <KDateTime>
 
 #include <Plasma/DataContainer>
 
 #include <rtm/rtm.h> 
 #include <rtm/session.h>
 
-const QString RtmEngine::apiKey = "54c4c997b087ba69b5700efd49f774d4";
-const QString RtmEngine::sharedSecret = "85dc83023257714a";
+const QString RtmEngine::apiKey = "631e881f0e5671d237c1a2a0a64d5b98";
+const QString RtmEngine::sharedSecret = "a1c48d8944bce414";
+
 
 RtmEngine::RtmEngine(QObject* parent, const QVariantList& args)
     : Plasma::DataEngine(parent, args),
@@ -120,23 +117,23 @@ bool RtmEngine::updateSourceEvent(const 
     return true;
   }
   else if (source.startsWith("Lists")) {
-    ListsSource *listssoruce = dynamic_cast<ListsSource*>(containerForSource(source)); 
+    ListsSource *listssoruce = static_cast<ListsSource*>(containerForSource(source)); 
     listssoruce->refresh();
     return true;
   }
   else if (source.startsWith("Tasks")) {
-    TasksSource *taskssource = dynamic_cast<TasksSource*>(containerForSource(source)); 
+    TasksSource *taskssource = static_cast<TasksSource*>(containerForSource(source)); 
     taskssource->refresh();
     return true;
   }
   else if (source.startsWith("List:")) {
-    ListSource *listsource = dynamic_cast<ListSource*>(containerForSource(source)); 
+    ListSource *listsource = static_cast<ListSource*>(containerForSource(source)); 
     if (listsource)
       listsource->update();
     return true;
   }
   else if (source.startsWith("Task:")) {
-    TaskSource *tasksource = dynamic_cast<TaskSource*>(containerForSource(source)); 
+    TaskSource *tasksource = static_cast<TaskSource*>(containerForSource(source)); 
     if (tasksource)
       tasksource->update();
     return true;
diff -rupN 4.3-kdeplasma-addons/dataengines/rememberthemilk/rtmengine.h trunk-kdeplasma-addons/dataengines/rememberthemilk/rtmengine.h
--- 4.3-kdeplasma-addons/dataengines/rememberthemilk/rtmengine.h	2009-10-10 11:00:19.000000000 -0400
+++ trunk-kdeplasma-addons/dataengines/rememberthemilk/rtmengine.h	2009-10-10 11:06:30.000000000 -0400
@@ -24,9 +24,11 @@
 
 #include <rtm/rtm.h>
 
-class TasksSource;
-class ListsSource;
 class TaskSource;
+namespace RTM
+{
+  class Session;
+}
 
 class RtmEngine : public Plasma::DataEngine
 {
diff -rupN 4.3-kdeplasma-addons/dataengines/rememberthemilk/taskservice.cpp trunk-kdeplasma-addons/dataengines/rememberthemilk/taskservice.cpp
--- 4.3-kdeplasma-addons/dataengines/rememberthemilk/taskservice.cpp	2009-10-10 11:00:19.000000000 -0400
+++ trunk-kdeplasma-addons/dataengines/rememberthemilk/taskservice.cpp	2009-10-10 11:06:30.000000000 -0400
@@ -21,7 +21,6 @@
 #include "tasksource.h"
 
 #include <rtm/session.h>
-#include <QMetaClassInfo>
 
 TaskService::TaskService(RTM::Session* session, RTM::Task* task, TaskSource* parent)
   : Plasma::Service(parent)
diff -rupN 4.3-kdeplasma-addons/dataengines/rememberthemilk/taskservice.h trunk-kdeplasma-addons/dataengines/rememberthemilk/taskservice.h
--- 4.3-kdeplasma-addons/dataengines/rememberthemilk/taskservice.h	2009-10-10 11:00:19.000000000 -0400
+++ trunk-kdeplasma-addons/dataengines/rememberthemilk/taskservice.h	2009-10-10 11:06:30.000000000 -0400
@@ -26,7 +26,6 @@
 class TaskSource;
 
 namespace RTM {
-  class Request;
   class Session;
   class Task;
 }
diff -rupN 4.3-kdeplasma-addons/dataengines/rememberthemilk/tasksservice.h trunk-kdeplasma-addons/dataengines/rememberthemilk/tasksservice.h
--- 4.3-kdeplasma-addons/dataengines/rememberthemilk/tasksservice.h	2009-10-10 11:00:19.000000000 -0400
+++ trunk-kdeplasma-addons/dataengines/rememberthemilk/tasksservice.h	2009-10-10 11:06:30.000000000 -0400
@@ -24,7 +24,6 @@
 #include <Plasma/ServiceJob>
 
 namespace RTM {
-  class Request;
   class Session;
 }
 
diff -rupN 4.3-kdeplasma-addons/dataengines/rememberthemilk/taskssource.cpp trunk-kdeplasma-addons/dataengines/rememberthemilk/taskssource.cpp
--- 4.3-kdeplasma-addons/dataengines/rememberthemilk/taskssource.cpp	2009-10-10 11:00:19.000000000 -0400
+++ trunk-kdeplasma-addons/dataengines/rememberthemilk/taskssource.cpp	2009-10-10 11:06:30.000000000 -0400
@@ -23,10 +23,7 @@
 #include <rtm/task.h>
 #include <rtm/session.h>
 #include <rtm/request.h>
-#include <QNetworkReply>
 
-#include <QTimer>
-#include <QBuffer>
 
 #include <KIO/Job>
 #include "rtmengine.h"
diff -rupN 4.3-kdeplasma-addons/dataengines/rememberthemilk/taskssource.h trunk-kdeplasma-addons/dataengines/rememberthemilk/taskssource.h
--- 4.3-kdeplasma-addons/dataengines/rememberthemilk/taskssource.h	2009-10-10 11:00:19.000000000 -0400
+++ trunk-kdeplasma-addons/dataengines/rememberthemilk/taskssource.h	2009-10-10 11:06:30.000000000 -0400
@@ -24,8 +24,6 @@
 #include <rtm/task.h>
 #include <rtm/list.h>
 
-#include <QList>
-#include <QPair>
 #include <QTimer>
 
 #include <Plasma/DataContainer>
@@ -34,7 +32,6 @@
 
 // forward declarations
 class TaskSource;
-class QNetworkReply;
 class TasksSource;
 class RtmEngine;
 
diff -rupN 4.3-kdeplasma-addons/libs/rtm/CMakeLists.txt trunk-kdeplasma-addons/libs/rtm/CMakeLists.txt
--- 4.3-kdeplasma-addons/libs/rtm/CMakeLists.txt	2009-10-10 11:00:34.000000000 -0400
+++ trunk-kdeplasma-addons/libs/rtm/CMakeLists.txt	2009-10-10 11:06:48.000000000 -0400
@@ -23,7 +23,7 @@ set( rtm_LIB_SRCS 
 
 kde4_add_library(rtm SHARED ${rtm_LIB_SRCS})
 set_target_properties(rtm PROPERTIES VERSION ${GENERIC_LIB_VERSION} SOVERSION ${GENERIC_LIB_SOVERSION})
-target_link_libraries(rtm ${KDE4_KDEUI_LIBS} ${KDE4_KIO_LIBS} ${KDE4_KHTML_LIBS})
+target_link_libraries(rtm ${KDE4_KDEUI_LIBS} ${KDE4_KIO_LIBS} ${QT_QTWEBKIT_LIBRARY})
 
 
 #### FOR 4.3 we are not including the library as a stable platform, hence no exporting of headers
diff -rupN 4.3-kdeplasma-addons/libs/rtm/Messages.sh trunk-kdeplasma-addons/libs/rtm/Messages.sh
--- 4.3-kdeplasma-addons/libs/rtm/Messages.sh	1969-12-31 19:00:00.000000000 -0500
+++ trunk-kdeplasma-addons/libs/rtm/Messages.sh	2009-10-10 11:06:48.000000000 -0400
@@ -0,0 +1,2 @@
+#! /usr/bin/env bash
+$XGETTEXT *.cpp -o $podir/librtm.pot
diff -rupN 4.3-kdeplasma-addons/libs/rtm/auth.cpp trunk-kdeplasma-addons/libs/rtm/auth.cpp
--- 4.3-kdeplasma-addons/libs/rtm/auth.cpp	2009-10-10 11:00:34.000000000 -0400
+++ trunk-kdeplasma-addons/libs/rtm/auth.cpp	2009-10-10 11:06:48.000000000 -0400
@@ -24,20 +24,20 @@
 #include <QNetworkAccessManager>
 #include <QNetworkRequest>
 #include <QCoreApplication>
+#include <QWebView>
+#include <QPushButton>
 
 #include <KDebug>
 #include <KIO/NetAccess>
-#include <KHTMLPart>
-#include <KHTMLView>
-#include <DOM/HTMLInputElement>
-#include <DOM/HTMLDocument>
+#include <QVBoxLayout>
+#include <KLocale>
 
 RTM::Auth::Auth(RTM::Permissions permissions, const QString& apiKey, const QString& sharedSecret)
-  : authPage(0)
+  : frobRequest(0),
+  tokenRequest(0)
 {
-  frobRequest = new RTM::Request("rtm.auth.getFrob", apiKey, sharedSecret);
-  tokenRequest = new RTM::Request("rtm.auth.getToken", apiKey, sharedSecret);
   arguments.insert("perms", getTextPermissions(permissions));
+  this->apiKey = apiKey;
   this->sharedSecret = sharedSecret;
   arguments.insert("api_key", apiKey);
   m_state = RTM::Mutable;
@@ -48,89 +48,62 @@ RTM::Auth::~Auth() {
   tokenRequest->deleteLater();
 }
 
-void RTM::Auth::login(const QString& authUrl, const QString& username, const QString& password) {
-  kDebug() << "Starting Login for user: " << username;
-  KIO::Job *job = KIO::get(KUrl(authUrl), KIO::NoReload, KIO::HideProgressInfo);
-  QByteArray data;
-  KIO::NetAccess::synchronousRun(job, 0, &data);
-  job->deleteLater();
-  
-  authPage = new KHTMLPart();
-  authPage->setJScriptEnabled(false);
-  authPage->begin(KUrl(authUrl));
-  authPage->write(data.constData());
-  authPage->end();
-  
-  //authPage->view()->resize(500, 400);
-  //authPage->widget()->show();
-  
-  m_username = username;
-  m_password = password;
-  
-  connect(authPage, SIGNAL(completed()), SLOT(pageLoaded()));
-  connect(authPage->browserExtension(), SIGNAL(openUrlRequestDelayed(const KUrl&, const KParts::OpenUrlArguments&, const KParts::BrowserArguments&)),
-          SLOT(pageLoadingReq(const KUrl&, const KParts::OpenUrlArguments&, const KParts::BrowserArguments&))); 
-          
-  authCount = 0;
-}
-
-void RTM::Auth::pageLoaded() {
-  kDebug() << "Stage " << authCount;
-  
-  if (authCount >= 2) {
-    disconnect(authPage);
-    continueAuthForToken();
-    return;
-  }
-    
-  DOM::HTMLInputElement auth = authPage->htmlDocument().getElementById("authorize_yes");
-  if (!auth.isNull()) {
-    kDebug() << "Entering Stage 2";
-    authCount = 2;
-    auth.click();
-    return;
-  }
+
+void RTM::Auth::showLoginWebpage()
+{
+  if (frobRequest)
+    frobRequest->deleteLater();
   
-  DOM::HTMLInputElement uname = authPage->htmlDocument().getElementById("username");
-  DOM::HTMLInputElement pword = authPage->htmlDocument().getElementById("password");
-  DOM::HTMLFormElement form = authPage->htmlDocument().getElementById("loginform");
-  
-  if (uname.isNull()) {
-    authCount = 2;
-    disconnect(authPage);
-    authPage->deleteLater();
-    continueAuthForToken();
-    return;
-  }
+  frobRequest = new RTM::Request("rtm.auth.getFrob", apiKey, sharedSecret);
+  connect(frobRequest, SIGNAL(replyReceived(RTM::Request*)), SLOT(showLoginWindowInternal(RTM::Request*)));
+  frobRequest->sendRequest();
+}
+
+
+void RTM::Auth::showLoginWindowInternal(RTM::Request *rawReply)
+{
+  QString reply = rawReply->data();
+  //QString reply = rawReply->readAll(); //FIXME: I have no idea why this line isn't working?
+  frob = reply.remove(0, reply.indexOf("<frob>")+6);
+  frob.truncate(frob.indexOf("</frob>"));
+  kDebug() << "Frob: " << frob;
+  arguments.insert("frob", frob);
   
-  uname.setValue(m_username);
-  pword.setValue(m_password);
   
-  kDebug() << "Entering Stage 1";
-    
-  form.submit();
+  QWidget *authWidget = new QWidget();
+  QVBoxLayout *layout = new QVBoxLayout(authWidget);
+  QPushButton *button = new QPushButton(authWidget);
+  QWebView *authPage  = new QWebView(authWidget);
+  
+  button->setText(i18n("Click here after you've logged in and authorized the applet"));
+
+  authPage->setUrl(getAuthUrl());
   
-  authCount = 1;
+  authPage->resize(800, 600);
+  authPage->scroll(0, 200);
+  
+  layout->addWidget(authPage);
+  layout->addWidget(button);
+  
+ 
+  connect(button, SIGNAL(clicked(bool)), authWidget, SLOT(hide()));
+  connect(button, SIGNAL(clicked(bool)), authWidget, SLOT(deleteLater()));
+  connect(button, SIGNAL(clicked(bool)), SLOT(pageClosed())); // Last because it takes more time.
+  
+  authWidget->show();
 }
 
-void RTM::Auth::pageLoadingReq(const KUrl& url, const KParts::OpenUrlArguments& args, const KParts::BrowserArguments& browserArgs)
-{
-  kDebug() << url;
-  authPage->setArguments(args);
-  authPage->browserExtension()->setBrowserArguments(browserArgs);
-  authPage->openUrl(url);
+
+void RTM::Auth::pageClosed() {
+  continueAuthForToken();
 }
 
 QString RTM::Auth::getAuthUrl() {
-  QString reply = frobRequest->sendSynchronousRequest();
-  frob = reply.remove(0, reply.indexOf("<frob>")+6);
-  frob.truncate(frob.indexOf("</frob>"));
-  kDebug() << "Frob: " << frob;
-  arguments.insert("frob", frob);
+  if (frob.isEmpty())
+    kWarning() << "Warning, Frob is EMPTY";
   return requestUrl();
 }
 
-
 QString RTM::Auth::getTextPermissions(RTM::Permissions permissions)
 {
   QString textPermissions;
@@ -156,12 +129,14 @@ QString RTM::Auth::getTextPermissions(RT
 }
 
 QString RTM::Auth::requestUrl() {
-  kDebug() << "RTM::Auth::getRequestUrl()";
+  kDebug() << "RTM::Auth::getRequestUrl()" << m_state << RTM::Mutable;
   switch(m_state) {
     case RTM::Mutable:
       sign();
       break;
     case RTM::Hashed:
+      unsign();
+      sign();
       break;
     case RTM::RequestSent:
       break;
@@ -176,11 +151,24 @@ QString RTM::Auth::requestUrl() {
 
 void RTM::Auth::continueAuthForToken()
 {
+  kDebug() << "Token Time";
+  if (tokenRequest)
+    tokenRequest->deleteLater();
+  
+  tokenRequest = new RTM::Request("rtm.auth.getToken", apiKey, sharedSecret);
   tokenRequest->addArgument("frob", arguments.value("frob"));
-  QString reply = tokenRequest->sendSynchronousRequest();
+  connect(tokenRequest, SIGNAL(replyReceived(RTM::Request*)), SLOT(tokenResponse(RTM::Request*)));
+  tokenRequest->sendRequest();
+}
+
 
+void RTM::Auth::tokenResponse(RTM::Request* response)
+{
+  QString reply = response->data();
+  kDebug() << "Reply: " << reply;
   QString token = reply.remove(0, reply.indexOf("<token>")+7);
   token.truncate(token.indexOf("</token>"));
   kDebug() << "Token: " << token;
   emit tokenReceived(token);
 }
+
diff -rupN 4.3-kdeplasma-addons/libs/rtm/auth.h trunk-kdeplasma-addons/libs/rtm/auth.h
--- 4.3-kdeplasma-addons/libs/rtm/auth.h	2009-10-10 11:00:34.000000000 -0400
+++ trunk-kdeplasma-addons/libs/rtm/auth.h	2009-10-10 11:06:48.000000000 -0400
@@ -30,8 +30,8 @@
 
 #include "request.h"
 
+class QWebView;
 
-class KHTMLPart;
 
 namespace RTM {
 
@@ -40,7 +40,7 @@ class RTM_EXPORT Auth : public Request
 Q_OBJECT
   public:
     Auth(RTM::Permissions permissions, const QString &apiKey, const QString &sharedSecret);
-    void login(const QString &authUrl, const QString &username, const QString &password);
+    void showLoginWebpage();
     QString getAuthUrl();
     void continueAuthForToken();
 
@@ -49,20 +49,19 @@ Q_OBJECT
     QString getTextPermissions(RTM::Permissions permissions);
     QString requestUrl();
     QString frob;
+    QString apiKey;
     Request *frobRequest;
     Request *tokenRequest;
-    KHTMLPart* authPage;
-    QString m_username;
-    QString m_password;
-    int authCount;
 
   signals:
     void authUrlReady(QString authUrl);
     void tokenReceived(QString token);
 
   protected slots:
-    void pageLoaded();
-    void pageLoadingReq(const KUrl& url, const KParts::OpenUrlArguments& args, const KParts::BrowserArguments& browserArgs);
+    void pageClosed();
+    void showLoginWindowInternal(RTM::Request* rawReply);
+public slots:
+    void tokenResponse(RTM::Request*);
 };
 
 } // Namespace RTM
diff -rupN 4.3-kdeplasma-addons/libs/rtm/request.cpp trunk-kdeplasma-addons/libs/rtm/request.cpp
--- 4.3-kdeplasma-addons/libs/rtm/request.cpp	2009-10-10 11:00:34.000000000 -0400
+++ trunk-kdeplasma-addons/libs/rtm/request.cpp	2009-10-10 11:06:48.000000000 -0400
@@ -119,12 +119,18 @@ void RTM::Request::sign() {
   m_state = RTM::Hashed;
 }
 
+void RTM::Request::unsign() {
+  arguments.remove("api_sig");
+}
+
 QString RTM::Request::requestUrl()
 {
   switch(m_state) {
     case RTM::Mutable:
       sign();
     case RTM::Hashed:
+      unsign();
+      sign();
       break;
     case RTM::RequestSent:
       break;
diff -rupN 4.3-kdeplasma-addons/libs/rtm/request.h trunk-kdeplasma-addons/libs/rtm/request.h
--- 4.3-kdeplasma-addons/libs/rtm/request.h	2009-10-10 11:00:34.000000000 -0400
+++ trunk-kdeplasma-addons/libs/rtm/request.h	2009-10-10 11:06:48.000000000 -0400
@@ -61,6 +61,7 @@ Q_OBJECT
   protected:
     Request() {}
     void sign();
+    void unsign();
 
     QMap<QString,QString> arguments;
     QString m_response;
diff -rupN 4.3-kdeplasma-addons/libs/rtm/rtm.h trunk-kdeplasma-addons/libs/rtm/rtm.h
--- 4.3-kdeplasma-addons/libs/rtm/rtm.h	2009-10-10 11:00:34.000000000 -0400
+++ trunk-kdeplasma-addons/libs/rtm/rtm.h	2009-10-10 11:06:48.000000000 -0400
@@ -25,14 +25,12 @@
 #include <QHash>
 #include <QString>
 
-class QStringList;
 class QString;
-class QByteArray;
 
 namespace RTM {
   enum State { Mutable, Hashed, RequestSent, RequestReceived };
   enum Permissions { None , Read, Write, Delete };
-  const QString baseAuthUrl = "http://www.rememberthemilk.com/services/auth/?";
+  const QString baseAuthUrl = "https://www.rememberthemilk.com/services/auth/?";
   const QString baseMethodUrl = "http://api.rememberthemilk.com/services/rest/?";
 
   /** Timlines are unsigned longs that map to a "session" in which
@@ -48,15 +46,11 @@ namespace RTM {
   typedef QString Tag;
 
   class List;
-  class Session;
   class Request;
   class Auth;
   class Task;
-  class StaticTask;
   class Note;
-  class Object;
 
-  class TasksReader;
 
   class Location
   {
diff -rupN 4.3-kdeplasma-addons/libs/rtm/session.cpp trunk-kdeplasma-addons/libs/rtm/session.cpp
--- 4.3-kdeplasma-addons/libs/rtm/session.cpp	2009-10-10 11:00:34.000000000 -0400
+++ trunk-kdeplasma-addons/libs/rtm/session.cpp	2009-10-10 11:06:48.000000000 -0400
@@ -29,7 +29,6 @@
 #include <QStringList>
 #include <QByteArray>
 #include <QCryptographicHash>
-#include <QHttp>
 #include <QBuffer>
 #include <QCoreApplication>
 
@@ -47,6 +46,9 @@ RTM::Session::Session(QString apiKey, QS
   d->apiKey = apiKey;
   d->sharedSecret = sharedSecret;
   d->permissions = permissions;
+  
+  connect(this, SIGNAL(tokenCheck(bool)), SLOT(handleValidToken(bool)));
+  
   setToken(token);
 }
 
@@ -55,14 +57,15 @@ RTM::Session::~Session()
   delete d;
 }
 
-void RTM::Session::login(const QString& username, const QString& password) {
+void RTM::Session::showLoginWindow() {
   //FIXME: What happens when auth wasn't created?
   if (!d->auth)
     d->auth = new RTM::Auth(d->permissions, d->apiKey, d->sharedSecret);
-  d->auth->login(d->authUrl, username, password);
+  d->auth->showLoginWebpage();
 }
 
 
+
 QString RTM::Session::getAuthUrl() const {
   return d->authUrl;
 }
@@ -94,22 +97,27 @@ QString RTM::Session::sharedSecret() con
 RTM::Timeline RTM::Session::getTimeline() const {
   return d->timeline;
 }
-
+  
 QString RTM::Session::token() const {
   return d->token;
 }
 
 void RTM::Session::setToken(const QString& token)
 {
+  d->token = token;
+  d->tasks.clear(); //FIXME: Leak? Tasks/Lists are pointers.
+  d->lists.clear();
+  
+  checkToken();
+}
+
+void RTM::Session::handleValidToken(bool valid)
+{
   if (d->auth) {
     d->auth->deleteLater();
     d->auth = 0;
   }
-    
-  d->token = token;
-  d->tasks.clear(); //FIXME: Leak? Tasks/Lists are pointers.
-  d->lists.clear();
-  if (!checkToken()) {
+  if (!valid) {
     d->token.clear();
     d->auth = new RTM::Auth(d->permissions, d->apiKey, d->sharedSecret);
     d->authUrl = d->auth->getAuthUrl();
@@ -122,6 +130,7 @@ void RTM::Session::setToken(const QStrin
   }
 }
 
+
 void RTM::Session::setTimeline(const RTM::Timeline& timeline) {
   d->timeline = timeline;
 }
@@ -131,37 +140,53 @@ void RTM::Session::continueAuthForToken(
   d->auth->continueAuthForToken();
 }
 
-bool RTM::Session::checkToken() {
-  if (d->token.isEmpty())
-    return false;
-  RTM::Request tokenRequest("rtm.auth.checkToken", d->apiKey, d->sharedSecret);
-  tokenRequest.addArgument("auth_token", d->token);
-  QString reply = tokenRequest.sendSynchronousRequest();
+void RTM::Session::checkToken() {
+  if (d->token.isEmpty()) {
+    emit tokenCheck(false);
+    return;
+  }
+  RTM::Request *tokenRequest = new RTM::Request("rtm.auth.checkToken", d->apiKey, d->sharedSecret);
+  tokenRequest->addArgument("auth_token", d->token);
+  connect(tokenRequest, SIGNAL(replyReceived(RTM::Request*)), SLOT(tokenCheckReply(RTM::Request*)));
+  connect(tokenRequest, SIGNAL(replyReceived(RTM::Request*)), tokenRequest, SLOT(deleteLater()));
+  tokenRequest->sendRequest();
+}
+
 
+void RTM::Session::tokenCheckReply(RTM::Request* response)
+{
+  QString reply = response->data();
+  
   if (!reply.contains(d->token)) {
     kDebug() << "Failed Token Check: " << reply;
     emit tokenCheck(false);
-    return false;
   }
   kDebug() << "Successful Token Check: " << reply;
   emit tokenCheck(true);
-  return true;
 }
 
-RTM::Timeline RTM::Session::createTimeline() {
+
+void RTM::Session::createTimeline() {
   RTM::Request *request = new RTM::Request("rtm.timelines.create", d->apiKey, d->sharedSecret);
   request->addArgument("auth_token", d->token);
-  QString reply = request->sendSynchronousRequest();
+  connect(request, SIGNAL(replyReceived(RTM::Request*)), SLOT(timelineReply(RTM::Request*)));
+  connect(request, SIGNAL(replyReceived(RTM::Request*)), request, SLOT(deleteLater()));
+  request->sendRequest();
+}
+
+void RTM::Session::timelineReply(RTM::Request* response)
+{
+  QString reply = response->data();
   QString timeline = reply.remove(0, reply.indexOf("<timeline>")+10);
   timeline.truncate(timeline.indexOf("</timeline>"));
   kDebug() << "Timeline: " << timeline;
-  request->deleteLater();
   d->timeline = timeline.toLong();
   d->lastRefresh = QDateTime();
-  return getTimeline();
+  emit timelineCreated(getTimeline());
 }
 
 
+
 void RTM::Session::handleResponse()
 {
     //TODO: Move data from buffer to response, then clear buffer.
diff -rupN 4.3-kdeplasma-addons/libs/rtm/session.h trunk-kdeplasma-addons/libs/rtm/session.h
--- 4.3-kdeplasma-addons/libs/rtm/session.h	2009-10-10 11:00:34.000000000 -0400
+++ trunk-kdeplasma-addons/libs/rtm/session.h	2009-10-10 11:06:48.000000000 -0400
@@ -32,7 +32,6 @@ namespace RTM {
   class TasksReader;
 }
 
-class QNetworkReply;
 
 /** @file
  * This file is part of librtm. It defines
@@ -60,7 +59,7 @@ Q_OBJECT
     virtual ~Session();
 
     QString getAuthUrl() const;
-    void login(const QString &username, const QString &password);
+    void showLoginWindow();
     bool authenticated() const;
 
     RTM::Request* request(const QString& method);
@@ -69,9 +68,9 @@ Q_OBJECT
 
     void setTimeline(const RTM::Timeline& timeline);
     RTM::Timeline getTimeline() const;
-    RTM::Timeline createTimeline();
+    void createTimeline();
 
-    bool checkToken();
+    void checkToken();
 
     QString apiKey() const;
     QString sharedSecret() const;
@@ -95,10 +94,15 @@ Q_OBJECT
     void continueAuthForToken();
     void addTask(const QString &task, RTM::ListId listId);
     RTM::Task* createTaskFromString(const QString& task);
+    void tokenCheckReply(RTM::Request*);
+    void handleValidToken(bool);
+    void timelineReply(RTM::Request*);
 
   Q_SIGNALS:
     void tokenReceived(const QString& token);
     void tokenCheck(bool success);
+    
+    void timelineCreated(RTM::Timeline timeline);
 
     void taskChanged(RTM::Task* task);
     void listChanged(RTM::List* list);
diff -rupN 4.3-kdeplasma-addons/libs/rtm/session_p.h trunk-kdeplasma-addons/libs/rtm/session_p.h
--- 4.3-kdeplasma-addons/libs/rtm/session_p.h	2009-10-10 11:00:34.000000000 -0400
+++ trunk-kdeplasma-addons/libs/rtm/session_p.h	2009-10-10 11:06:48.000000000 -0400
@@ -32,7 +32,6 @@
 #include <QString>
 #include <QByteArray>
 #include <QCryptographicHash>
-#include <QHttp>
 #include <QBuffer>
 #include <QCoreApplication>
 #include <QDomDocument>
@@ -86,7 +85,7 @@ class RTM::SessionPrivate {
   void smartListReply(RTM::Request* reply) {
     QStringList parts = reply->requestUrl().split("&");
     RTM::ListId id = 0;
-    foreach(QString part, parts)
+    foreach(const QString &part, parts)
       if (part.contains("rtm_internal_list_id"))
         id = part.split("=").last().toLongLong();
       
diff -rupN 4.3-kdeplasma-addons/libs/rtm/task.h trunk-kdeplasma-addons/libs/rtm/task.h
--- 4.3-kdeplasma-addons/libs/rtm/task.h	2009-10-10 11:00:34.000000000 -0400
+++ trunk-kdeplasma-addons/libs/rtm/task.h	2009-10-10 11:06:48.000000000 -0400
@@ -35,6 +35,7 @@
 namespace RTM {
 
 class TaskPrivate;
+class Session;
 
 class RTM_EXPORT Task : public QObject
 {
diff -rupN 4.3-kdeplasma-addons/libs/rtm/tests/login.cpp trunk-kdeplasma-addons/libs/rtm/tests/login.cpp
--- 4.3-kdeplasma-addons/libs/rtm/tests/login.cpp	2009-10-10 11:00:34.000000000 -0400
+++ trunk-kdeplasma-addons/libs/rtm/tests/login.cpp	2009-10-10 11:06:48.000000000 -0400
@@ -5,23 +5,17 @@
 #include <KCmdLineArgs>
 #include <KDebug>
 
-#include <KIO/Job>
-#include <KIO/NetAccess>
-
-#include <KHTMLPart>
-#include <KHTMLView>
-#include <DOM/HTMLInputElement>
-#include <DOM/HTMLDocument>
-
-int main(int argc, char* argv[]) {
+int main(int argc, char* argv[]) { 
   KAboutData about("login", 0, ki18n("RTM Login Test"), "1.0", ki18n("A Simple RTM Login Test"),
                   KAboutData::License_GPL, ki18n("(C) 2009 Andrew Stromme"), KLocalizedString(), 0, "astromme@chatonka.com");
   KCmdLineArgs::init(argc, argv, &about);
+  
   KApplication app;
-
-  RTM::Session *session = new RTM::Session("myapikey", "mysharedsecret", RTM::Delete, QString(), &app);
   
-  session->login("myuser", "mypass");
+  RTM::Session *session = new RTM::Session("myapikey", "mysharedsecret", RTM::Delete, QString(), &app);
+  
+  session->showLoginWindow();
   
-  app.exec();
+  return app.exec();
 }
+
diff -rupN 4.3-kdeplasma-addons/libs/rtm/tests/refresh.cpp trunk-kdeplasma-addons/libs/rtm/tests/refresh.cpp
--- 4.3-kdeplasma-addons/libs/rtm/tests/refresh.cpp	2009-10-10 11:00:34.000000000 -0400
+++ trunk-kdeplasma-addons/libs/rtm/tests/refresh.cpp	2009-10-10 11:06:48.000000000 -0400
@@ -45,7 +45,7 @@ int main(int argc, char* argv[]) {
                     
   RTM::Session *session = new RTM::Session("myapikey", "mysharedsecret", RTM::Delete, QString(), &app);
   
-  session->login("myuser", "mypass");
+  session->showLoginWindow();
   
   Refresher refresher;
   refresher.session = session;
diff -rupN 4.3-kdeplasma-addons/libs/rtm/xmlreaders.h trunk-kdeplasma-addons/libs/rtm/xmlreaders.h
--- 4.3-kdeplasma-addons/libs/rtm/xmlreaders.h	2009-10-10 11:00:34.000000000 -0400
+++ trunk-kdeplasma-addons/libs/rtm/xmlreaders.h	2009-10-10 11:06:48.000000000 -0400
@@ -23,9 +23,7 @@
 namespace RTM {
   class Request;
   class Session;
-  class SessionPrivate;
   class Task;
-  class TaskPrivate;
 }
 
 #include <QXmlStreamReader>
