Index: contacts/contactsrunner.cpp
===================================================================
--- kdeplasma-addons-4.3.0/runners/contacts/contactsrunner.cpp	(revision 979287)
+++ kdeplasma-addons-4.3.0/runners/contacts/contactsrunner.cpp	(working copy)
@@ -47,6 +47,7 @@
     setIgnoredTypes(Plasma::RunnerContext::Directory | Plasma::RunnerContext::File |
                     Plasma::RunnerContext::NetworkLocation);
     addSyntax(Plasma::RunnerSyntax(":q:", i18n("Finds people in your address book matching :q:.")));
+    addSyntax(Plasma::RunnerSyntax(i18nc("list of all people in address book", "list contacts"), i18n("List all people in your address book.")));
 }
 
 ContactsRunner::~ContactsRunner()
@@ -69,10 +69,18 @@
             return;
         }
 
-        const bool matchedName = a.realName().contains(term, Qt::CaseInsensitive);
-        bool matchedMail = false;
-
-        if (!matchedName) {
+         bool matchedName;
+         bool matchedMail;
+  
+         if (term.compare(i18nc("list of all people in address book", "list contacts"), Qt::CaseInsensitive) == 0) {
+             matchedName = true;
+             matchedMail = true;
+         } else {
+             matchedName = a.realName().contains(term, Qt::CaseInsensitive);
+             matchedMail = false;
+         }
+ 
+         if (!matchedName && !matchedMail) {
             // Name didn't match, so lets try the name portion of the email address
             const int indexOf = a.preferredEmail().indexOf(term, Qt::CaseInsensitive);
             matchedMail = indexOf > -1 && indexOf < a.preferredEmail().indexOf('@');
