--- kdeplasma-addons/applets/microblog/microblog.cpp	2009/10/10 18:54:44	1033678
+++ kdeplasma-addons/applets/microblog/microblog.cpp	2009/10/10 18:54:46	1033679
@@ -381,9 +381,9 @@
 
         //add the newbies
         int newCount = 0;
-        ulong maxId = m_lastTweet;
+        qulonglong maxId = m_lastTweet;
         foreach (const QString &id, data.keys()) {
-            ulong i = id.toULong();
+            qulonglong i = id.toULongLong();
             //kDebug() << i << m_lastTweet;
             if (i > m_lastTweet) {
                 newCount++;
@@ -464,7 +464,7 @@
     // Add more tweetWidgets if there are not enough
 
     if (m_tweetMap.count() > m_historySize) {
-        QMap<ulong, Plasma::DataEngine::Data>::iterator it = m_tweetMap.begin();
+        QMap<qulonglong, Plasma::DataEngine::Data>::iterator it = m_tweetMap.begin();
         while (it != m_tweetMap.end() && m_tweetMap.count() > m_historySize) {
             it = m_tweetMap.erase(it);
         }
@@ -521,7 +521,7 @@
     }
 
     int i = 0;
-    QMap<ulong, Plasma::DataEngine::Data>::iterator it = m_tweetMap.end();
+    QMap<qulonglong, Plasma::DataEngine::Data>::iterator it = m_tweetMap.end();
     while (it != m_tweetMap.begin()) {
         Plasma::DataEngine::Data &tweetData = *(--it);
         QString user = tweetData.value("User").toString();
--- kdeplasma-addons/applets/microblog/microblog.h	2009/10/10 18:54:44	1033678
+++ kdeplasma-addons/applets/microblog/microblog.h	2009/10/10 18:54:46	1033679
@@ -162,10 +162,10 @@
         QString m_curTimeline;
 
         QMap<QString, QPixmap> m_pictureMap;
-        QMap<ulong, Plasma::DataEngine::Data> m_tweetMap;
+        QMap<qulonglong, Plasma::DataEngine::Data> m_tweetMap;
         QList<Tweet> m_tweetWidgets;
 
-        ulong m_lastTweet;
+        qulonglong m_lastTweet;
         KWallet::Wallet *m_wallet;
         enum WalletWait { None=0, Read, Write };
         WalletWait m_walletWait;
