Index: kdeplasma-addons-4.6.1/applets/microblog/microblog.cpp
===================================================================
--- kdeplasma-addons-4.6.1.orig/applets/microblog/microblog.cpp	2011-02-25 22:49:51.000000000 +0000
+++ kdeplasma-addons-4.6.1/applets/microblog/microblog.cpp	2011-03-29 11:58:13.638063960 +0100
@@ -27,6 +27,8 @@
 #include <QGraphicsAnchorLayout>
 #include <QAction>
 #include <QTimer>
+#include <QFile>
+
 
 #include <KColorScheme>
 #include <KConfigDialog>
@@ -264,7 +266,7 @@
     QString serviceUrl = cg.readEntry("serviceUrl", "https://identi.ca/api/");
     QString username = cg.readEntry("username");
     QString password = KStringHandler::obscure(cg.readEntry("password"));
-    int historySize = cg.readEntry("historySize", 6);
+    int historySize = cg.readEntry("historySize", 10);
     int historyRefresh = cg.readEntry("historyRefresh", 5);
     bool includeFriends = cg.readEntry("includeFriends", true);
     bool changed = false;
@@ -339,6 +341,7 @@
 
     if (m_username.isEmpty()) {
         setAuthRequired(true);
+	downloadHistory();
     } else if (m_password.isEmpty()) {
         //kDebug() << "started, password is not in config file, trying wallet";
         m_walletWait = Read;
@@ -468,7 +471,9 @@
         //       to configure the widget;
 		m_password = KStringHandler::obscure(config().readEntry("password"));
 		if(m_password.isEmpty()){
-	        setConfigurationRequired(true, i18n("Your password is required."));
+                  if (!QFile::exists("/usr/bin/ubiquity")) { //don't show on live CD
+	            setConfigurationRequired(true, i18n("Your password is required."));
+	          }
 		}else{
 			//kDebug() << "reading from config";
         	//kDebug() << "failed to read password";
@@ -497,7 +502,9 @@
 
 void MicroBlog::setAuthRequired(bool required)
 {
-    setConfigurationRequired(required, i18n("Your account information is incomplete."));
+    if (!QFile::exists("/usr/bin/ubiquity")) {
+        setConfigurationRequired(required, i18n("Your account information is incomplete."));
+    }
     if (m_graphicsWidget) {
         m_statusEdit->setEnabled(!required);
     }
@@ -820,12 +827,18 @@
 {
    // kDebug() << "Inside downloadhistory";
     if (m_username.isEmpty() || m_password.isEmpty()) {
+      /*
 		//kDebug() << "BOOHYA got empty password";
         if (!m_curTimeline.isEmpty()) {
             m_engine->disconnectSource(m_curTimeline, this);
             m_engine->disconnectSource("Error:" + m_curTimeline, this);
         }
         return;
+      */
+	m_username = "kubuntu"; //default to show when unconfigured
+	m_headerFrame->hide();
+    } else {
+     m_headerFrame->show();
     }
 
     m_flash->flash(i18n("Refreshing timeline..."), -1);
@@ -838,7 +851,9 @@
 
         if (m_service) {
             KConfigGroup cg = m_service->operationDescription("auth");
-            cg.writeEntry("password", m_password);
+            if (!m_password.isEmpty()) {
+                cg.writeEntry("password", m_password);
+            }
             m_service->startOperationCall(cg);
         }
     }
@@ -878,8 +893,10 @@
         query = "Replies:%1@%2";
         break;
     default:
-        if(m_includeFriends) {
-            query = QString("TimelineWithFriends:%1@%2");
+        if (m_password.isEmpty()) {
+            query = QString("NoAuth:%1@%2");
+	} else if (m_includeFriends) {
+	    query = QString("TimelineWithFriends:%1@%2");
         } else {
             query = QString("Timeline:%1@%2");
         }
Index: kdeplasma-addons-4.6.1/dataengines/microblog/timelinesource.cpp
===================================================================
--- kdeplasma-addons-4.6.1.orig/dataengines/microblog/timelinesource.cpp	2011-02-25 22:49:51.000000000 +0000
+++ kdeplasma-addons-4.6.1/dataengines/microblog/timelinesource.cpp	2011-03-29 11:58:04.738063962 +0100
@@ -132,6 +132,9 @@
     case TimelineWithFriends:
         m_url = KUrl(m_serviceBaseUrl, "statuses/friends_timeline.xml");
         break;
+    case NoAuth:
+	m_url = KUrl(m_serviceBaseUrl, QString("statuses/user_timeline/%1.xml").arg(account.at(0)));
+	break;
     case Timeline:
     default:
         m_url = KUrl(m_serviceBaseUrl, "statuses/user_timeline.xml");
@@ -216,7 +219,7 @@
 
 void TimelineSource::update(bool forcedUpdate)
 {
-    if (m_job || (!account().isEmpty() && password().isEmpty() && !m_oauthToken.size())) {
+    if ( account() != "kubuntu" && (m_job || (!account().isEmpty() && password().isEmpty() && !m_oauthToken.size())) ) {
         // We are already performing a fetch, let's not bother starting over
         //kDebug() << "already updating....." << m_job << account().isEmpty() << password().isEmpty();
         return;
Index: kdeplasma-addons-4.6.1/dataengines/microblog/timelinesource.h
===================================================================
--- kdeplasma-addons-4.6.1.orig/dataengines/microblog/timelinesource.h	2011-02-25 22:49:51.000000000 +0000
+++ kdeplasma-addons-4.6.1/dataengines/microblog/timelinesource.h	2011-03-29 11:58:04.778063962 +0100
@@ -88,7 +88,8 @@
         TimelineWithFriends,
         Replies,
         DirectMessages,
-        Profile
+        Profile,
+	NoAuth
     };
 
     TimelineSource(const QString &who, RequestType requestType, QObject* parent);
Index: kdeplasma-addons-4.6.1/dataengines/microblog/twitterengine.cpp
===================================================================
--- kdeplasma-addons-4.6.1.orig/dataengines/microblog/twitterengine.cpp	2011-02-25 22:49:36.000000000 +0000
+++ kdeplasma-addons-4.6.1/dataengines/microblog/twitterengine.cpp	2011-03-29 11:58:04.808063962 +0100
@@ -30,6 +30,7 @@
 const QString TwitterEngine::timelinePrefix("Timeline:");
 const QString TwitterEngine::timelineWithFriendsPrefix("TimelineWithFriends:");
 const QString TwitterEngine::profilePrefix("Profile:");
+const QString TwitterEngine::noauthPrefix("NoAuth:");
 const QString TwitterEngine::repliesPrefix("Replies:");
 const QString TwitterEngine::messagesPrefix("Messages:");
 
@@ -53,7 +54,7 @@
 
     if (!name.startsWith(timelinePrefix) && !name.startsWith(timelineWithFriendsPrefix)
         && !name.startsWith(profilePrefix) && !name.startsWith(repliesPrefix)
-        && !name.startsWith(messagesPrefix)) {
+        && !name.startsWith(messagesPrefix) && !name.startsWith(noauthPrefix)) {
         return false;
     }
 
@@ -81,11 +82,10 @@
 //always returns false because everything is async
 bool TwitterEngine::updateSourceEvent(const QString &name)
 {
-    //kDebug() << name;
     //right now it only makes sense to do an update on timelines
     if (!name.startsWith(timelinePrefix) && !name.startsWith(timelineWithFriendsPrefix)
         && !name.startsWith(profilePrefix) && !name.startsWith(repliesPrefix)
-        && !name.startsWith(messagesPrefix)) {
+        && !name.startsWith(messagesPrefix) && !name.startsWith(noauthPrefix)) {
         return false;
     }
 
@@ -98,6 +98,9 @@
     } else if (name.startsWith(profilePrefix)) {
         requestType = TimelineSource::Profile;
         who.remove(profilePrefix);
+    } else if (name.startsWith(noauthPrefix)) {
+      requestType = TimelineSource::NoAuth;
+      who.remove(noauthPrefix);
     } else if (name.startsWith(repliesPrefix)) {
         requestType = TimelineSource::Replies;
         who.remove(repliesPrefix);
Index: kdeplasma-addons-4.6.1/dataengines/microblog/twitterengine.h
===================================================================
--- kdeplasma-addons-4.6.1.orig/dataengines/microblog/twitterengine.h	2011-02-25 22:49:36.000000000 +0000
+++ kdeplasma-addons-4.6.1/dataengines/microblog/twitterengine.h	2011-03-29 11:58:04.808063962 +0100
@@ -67,6 +67,7 @@
         static const QString timelinePrefix;
         static const QString timelineWithFriendsPrefix;
         static const QString profilePrefix;
+	static const QString noauthPrefix;
         static const QString repliesPrefix;
         static const QString messagesPrefix;
         ImageSource *m_imageSource;
