Index: kdeplasma-addons-4.4.80/applets/CMakeLists.txt
===================================================================
--- kdeplasma-addons-4.4.80.orig/applets/CMakeLists.txt	2010-05-20 07:44:27.000000000 -0400
+++ kdeplasma-addons-4.4.80/applets/CMakeLists.txt	2010-05-25 09:50:00.639462337 -0400
@@ -83,11 +83,7 @@
 if (Q_WS_X11)
    if (X11_Xrender_FOUND)
       set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${CMAKE_CURRENT_SOURCE_DIR}/kimpanel/cmake)
-      macro_optional_find_package(SCIM)
-      macro_log_feature(SCIM_FOUND "SCIM headers" "Multibyte input widget" "http://www.scim-im.org" FALSE "1.4" "Needed for the kimpanel multibyte input Plasmoid")
-      if (SCIM_FOUND)
          add_subdirectory(kimpanel)
-      endif ( SCIM_FOUND)
    endif (X11_Xrender_FOUND)
 endif (Q_WS_X11)
 
Index: kdeplasma-addons-4.4.80/applets/kimpanel/backend/CMakeLists.txt
===================================================================
--- kdeplasma-addons-4.4.80.orig/applets/kimpanel/backend/CMakeLists.txt	2009-11-30 19:35:52.000000000 -0500
+++ /dev/null	1970-01-01 00:00:00.000000000 +0000
@@ -1 +0,0 @@
-macro_optional_add_subdirectory(scim)
