Index: kdeplasma-addons-4.7.1/applets/CMakeLists.txt
===================================================================
--- kdeplasma-addons-4.7.1.orig/applets/CMakeLists.txt	2011-05-20 22:37:49.000000000 +0200
+++ kdeplasma-addons-4.7.1/applets/CMakeLists.txt	2011-09-04 11:29:14.000000000 +0200
@@ -87,11 +87,11 @@
 if (Q_WS_X11)
    if (X11_Xrender_FOUND)
       set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${CMAKE_CURRENT_SOURCE_DIR}/kimpanel/cmake)
-      macro_optional_find_package(SCIM)
-      macro_log_feature(SCIM_FOUND "SCIM headers" "Multibyte input widget" "http://www.scim-im.org" FALSE "1.4" "Needed for the kimpanel multibyte input Plasmoid")
-      if (SCIM_FOUND)
+#      macro_optional_find_package(SCIM)
+#      macro_log_feature(SCIM_FOUND "SCIM headers" "Multibyte input widget" "http://www.scim-im.org" FALSE "1.4" "Needed for the kimpanel multibyte input Plasmoid")
+#      if (SCIM_FOUND)
          add_subdirectory(kimpanel)
-      endif ( SCIM_FOUND)
+#      endif ( SCIM_FOUND)
    endif (X11_Xrender_FOUND)
 endif (Q_WS_X11)
 
Index: kdeplasma-addons-4.7.1/applets/kimpanel/backend/CMakeLists.txt
===================================================================
--- kdeplasma-addons-4.7.1.orig/applets/kimpanel/backend/CMakeLists.txt	2011-05-20 22:37:49.000000000 +0200
+++ kdeplasma-addons-4.7.1/applets/kimpanel/backend/CMakeLists.txt	2011-09-04 11:29:14.000000000 +0200
@@ -1 +1 @@
-macro_optional_add_subdirectory(scim)
+#macro_optional_add_subdirectory(scim)
