Index: kdeplasma-addons-4.1.85/applets/comic/fadingitem.cpp
===================================================================
--- kdeplasma-addons-4.1.85.orig/applets/comic/fadingitem.cpp	2008-12-31 01:25:37.000000000 +0000
+++ kdeplasma-addons-4.1.85/applets/comic/fadingitem.cpp	2008-12-31 01:25:50.000000000 +0000
@@ -61,7 +61,7 @@
     QPainter tempPainter( &temp );
     QColor color = Qt::black;
 
-    color.setAlphaF( qMin( mOpacity, 0.99 ) );
+    color.setAlphaF( qMin( double(mOpacity), 0.99 ) );
     tempPainter.setCompositionMode( QPainter::CompositionMode_DestinationIn );
     tempPainter.fillRect( mParent.rect(), color );
     painter->drawPixmap( QPoint( 0, 0 ), temp );
Index: kdeplasma-addons-4.1.85/applets/previewer/previewwidget.cpp
===================================================================
--- kdeplasma-addons-4.1.85.orig/applets/previewer/previewwidget.cpp	2008-12-31 02:09:49.000000000 +0000
+++ kdeplasma-addons-4.1.85/applets/previewer/previewwidget.cpp	2008-12-31 18:43:09.000000000 +0000
@@ -359,7 +359,7 @@
 void PreviewWidget::expandingSlot(qreal progress)
 {
     const int min = s_topBorder + bottomBorderHeight();
-    if (qFuzzyCompare(1.0, progress)) {
+    if (qFuzzyCompare(1.0, double(progress))) {
         m_animId = -1;
 
         if (m_closeStatus) {
@@ -374,7 +374,7 @@
             }
         }
     } else {
-        m_animationHeight = qMax(qreal(min),
+        m_animationHeight = qMax(double(min),
                                  min - 1 + ((size().height() - min)  * 
                                             (m_closeStatus ? 1.0 - progress : progress)));
     }
