--- kdeplasma-addons-4.4.0/applets/CMakeLists.txt	2009-12-17 16:38:19.000000000 +0000
+++ kdeplasma-addons-4.4.0/applets/CMakeLists.txt	2010-02-15 09:08:30.000000000 +0000
@@ -81,11 +81,7 @@
 if (Q_WS_X11)
    if (X11_Xrender_FOUND)
       set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${CMAKE_CURRENT_SOURCE_DIR}/kimpanel/cmake)
-      macro_optional_find_package(SCIM)
-      macro_log_feature(SCIM_FOUND "SCIM headers" "Multibyte input widget" "http://www.scim-im.org" FALSE "1.4" "Needed for the kimpanel multibyte input Plasmoid")
-      if (SCIM_FOUND)
          add_subdirectory(kimpanel)
-      endif ( SCIM_FOUND)
    endif (X11_Xrender_FOUND)
 endif (Q_WS_X11)
 
--- kdeplasma-addons-4.4.0/applets/kimpanel/backend/CMakeLists.txt	2009-12-01 00:35:52.000000000 +0000
+++ kdeplasma-addons-4.4.0/applets/kimpanel/backend/CMakeLists.txt	2010-02-15 09:14:01.000000000 +0000
@@ -1 +0,0 @@
-macro_optional_add_subdirectory(scim)
