Index: kdeplasma-addons-4.5.1/applets/frame/CMakeLists.txt
===================================================================
--- kdeplasma-addons-4.5.1.orig/applets/frame/CMakeLists.txt	2010-03-18 10:08:47.000000000 +0100
+++ kdeplasma-addons-4.5.1/applets/frame/CMakeLists.txt	2010-09-15 04:39:58.388879999 +0200
@@ -20,7 +20,7 @@
 
 if(HAVE_KEXIV2)
     include_directories( ${KEXIV2_INCLUDES} )
-    set_source_files_properties(picture.cpp PROPERTIES
+    set_source_files_properties(picture.cpp imageloader.cpp PROPERTIES
                                COMPILE_FLAGS -DHAVE_KEXIV2)
     target_link_libraries(plasma_applet_frame ${KDE4_PLASMA_LIBS} ${KDE4_KIO_LIBS} ${KDE4_KFILE_LIBS} ${KEXIV2_LIBRARIES} )
 else (HAVE_KEXIV2)
Index: kdeplasma-addons-4.5.1/applets/frame/imageloader.cpp
===================================================================
--- kdeplasma-addons-4.5.1.orig/applets/frame/imageloader.cpp	2010-04-22 10:04:27.000000000 +0200
+++ kdeplasma-addons-4.5.1/applets/frame/imageloader.cpp	2010-09-15 04:40:38.336880001 +0200
@@ -31,7 +31,7 @@
     m_path = path;
 }
 
-QImage ImageLoader::correctRotation(const QImage& tempImage)
+QImage ImageLoader::correctRotation(const QImage& tempImage, const QString &path)
 {
     QImage image = QImage();
     if (!tempImage.isNull()) {
@@ -81,7 +81,7 @@
 
 void ImageLoader::run()
 {
-    QImage img = correctRotation(QImage(m_path));
+    QImage img = correctRotation(QImage(m_path), m_path);
     emit loaded(img);
 }
 
Index: kdeplasma-addons-4.5.1/applets/frame/picture.cpp
===================================================================
--- kdeplasma-addons-4.5.1.orig/applets/frame/picture.cpp	2010-05-05 16:51:19.000000000 +0200
+++ kdeplasma-addons-4.5.1/applets/frame/picture.cpp	2010-09-15 04:39:58.388879999 +0200
@@ -139,7 +139,7 @@
         setPath(path);
     }
     
-    emit checkImageLoaded(ImageLoader::correctRotation(image));
+    emit checkImageLoaded(ImageLoader::correctRotation(image, path));
 }
 
 void Picture::checkImageLoaded(QImage newImage)
Index: kdeplasma-addons-4.5.1/applets/frame/imageloader.h
===================================================================
--- kdeplasma-addons-4.5.1/applets/frame/imageloader.h	2010/09/09 19:15:55	1173579
+++ kdeplasma-addons-4.5.1/applets/frame/imageloader.h	2010/09/09 19:17:29	1173580
@@ -32,7 +32,7 @@
 
 public:
     ImageLoader(const QString &path);
-    static QImage correctRotation(const QImage& tempImage);
+    static QImage correctRotation(const QImage& tempImage, const QString &path);
     void run();
 
 Q_SIGNALS:
