--- kdeplasma-addons-4.3.0/applets/opendesktop/opendesktop.cpp	2009-07-29 12:13:00.000000000 +0100
+++ kdeplasma-addons-4.3.0/applets/opendesktop/opendesktop.cpp	2009-08-06 17:05:03.000000000 +0100
@@ -81,6 +81,7 @@
 
 
     (void)graphicsWidget();
+    setMinimumSize(300, 300);
 }
 
 OpenDesktop::~OpenDesktop()
@@ -108,10 +109,11 @@
     }
     m_username = cg.readEntry("username", QString());
     m_displayedUser = m_username;
-    resize(420, 420);
+    resize(300, 300);
 
     if (m_username.isEmpty()) {
         setConfigurationRequired(true);
+        m_tabs->setCurrentIndex(2);
     } else {
         connectPerson(m_username);
         connectFriends(m_username);
@@ -159,7 +161,7 @@
 {
     if (!m_tabs) {
         m_tabs = new Plasma::TabBar(this);
-        m_tabs->setPreferredSize(300, 400);
+        m_tabs->setPreferredSize(300, 300);
         m_tabs->setMinimumSize(150, 200);
         m_tabs->setSizePolicy(QSizePolicy::Expanding, QSizePolicy::Expanding);
 
